src/components/Accordion/Accordion.tsx

```diff
@@ -8,12 +8,59 @@ import AccordionContent from './AccordionContent'
 import { DefaultBehavior } from '../../lib/accessibility'
 import { Accessibility } from '../../lib/accessibility/interfaces'
 
+export interface IAccordionPropsStrict {
+  /** An element type to render as (string or function). */
+  as?: any
+
+  /** Index of the currently active panel. */
+  activeIndex?: number[] | number
+
+  /** Primary content. */
+  children?: React.ReactNode
+
+  /** Additional classes. */
+  className?: string
+
+  /** Initial activeIndex value. */
+  defaultActiveIndex?: number[] | number
+
+  /** Only allow one panel open at a time. */
+  exclusive?: boolean
+
+  /**
+   * Called when a panel title is clicked.
+   *
+   * @param {SyntheticEvent} event - React's original SyntheticEvent.
+   * @param {object} data - All item props.
+   */
+  onTitleClick?: Function
+
+  /** Shorthand array of props for Accordion. */
+  panels?: {
+    content: React.ReactNode | object
+    title: React.ReactNode | object
+  }[]
+
+  /** Accessibility behavior if overridden by the user. */
+  accessibility?: object | Function
```

Comment:
`accessibility`, `styles`, `variables`, `className` are common for all components, shouldn't we extract them to separate interface. If so, then to how many?
Does the same apply to `children`?

Reply:
I was thinking about the same thing. For the children, they should be provided even from React itself so I am not sure if we should specify them here at all. The same should apply for the className as well, because it is React's property. For the rest, I would propose having the styles and variables in one interface, because for all themed components both would apply, and having the accessibility in other interface, because I am not sure if it is applicable to all components. We should think about the 'as' property as well. Having this said, I am kind a worry that we would end up with lot's of interfaces, which can become confusing, but at the same time I agree that it could be good for refactoring, if we need to add some common property or rename existing common property in the future for some reason.

Reply:
agree with the points mentioned above. However, my vote would rather be to do it after strict rtpes will be introduced for all components, so that at this point we will have clear picture around what are the general parts and what are the exceptions/edge cases.

Reply:
Got it! Will wait then for final approve for this PR in order for it to be merged.